test/i18n.js

```diff
@@ -48,3 +54,38 @@ test("get languages", async(tape) => {
   tape.same(i18n.getLanguages(), ["french", "english"]);
   tape.end();
 });
+
+test("Extend existing language", (tape) => {
+  i18n.extend("french", { welcome: "Bienvenue" });
+  tape.deepEqual(i18n.getToken("welcome"), "Bienvenue");
+  tape.end();
+});
+
+test("Add new language", async(tape) => {
+  i18n.extend("spanish", { welcome: "Bienvenido" });
+  await i18n.setLocalLang("spanish");
+  tape.deepEqual(i18n.getToken("welcome"), "Bienvenido");
+  tape.end();
+});
+
+test("extendFromSystemPath with non-existing languages directory", (tape) => {
+  tape.throws(() => {
+    i18n.extendFromSystemPath(Ki18nDir);
+  }, `The ${Ki18nDir} directory does not exist on this project.`);
+  tape.end();
+});
+
+test("extendFromSystemPath with existing languages directory", async(tape) => {
+  fs.mkdirSync(Ki18nDir);
+  fs.writeFileSync(path.join(Ki18nDir, "french.js"), JSON.stringify({ hello: "Bonjour" }));
+
+  i18n.extendFromSystemPath(Ki18nDir);
+
+  await i18n.setLocalLang("french");
+
+  tape.deepEqual(i18n.getToken("hello"), "Bonjour");
+
+  fs.unlinkSync(path.join(Ki18nDir, "french.js"));
+  fs.rmdirSync(Ki18nDir);
+  tape.end();
```

Comment:
You need to use a `try/finally` block to make sure you will always delete the created file.